src/Collection.php

```diff
@@ -833,24 +833,21 @@ public function mapReduce(JavascriptInterface $map, JavascriptInterface $reduce,
         }
 
         // Check if the out option is inline because we will want to coerce a primary read preference if not
-        if ( ! is_array($out) && ! is_object($out)) {
-            return false;
-        }
-
-        $out = (array) $out;
-
-        if (key($out) !== 'inline') {
+        if ($this->isOutInline($out)) {
+            $isOutInline = true;
             $options['readPreference'] = new ReadPreference(ReadPreference::RP_PRIMARY);
         }
 
         $server = $this->manager->selectServer($options['readPreference']);
 
-        // Set options if not set already
-        if ( ! isset($options['readConcern']) && ! ($this->readConcern->getLevel() === ReadConcern::MAJORITY) && \MongoDB\server_supports_feature($server, self::$wireVersionForReadConcern)) {
+        /* A "majority" read concern is not compatible with $out inline, so
```

Comment:
"$out" was a reference to the aggregation pipeline operator. "with inline output, so" may read better.